lib/Throttle.php

```diff
@@ -109,17 +110,56 @@ public function applyBruteForcePolicy($uid, $ip) {
 	 * @param string $ip
 	 * @return void
 	 */
-	public function clearSuspiciousAttemptsForUidIpCombination($uid, $ip) {
-		$this->attemptMapper->deleteSuspiciousAttemptsForUidIpCombination($uid, $ip);
+	public function clearFailedLoginAttemptsForUidIpCombination($uid, $ip) {
+		$this->loginAttemptMapper->deleteFailedLoginAttemptsForUidIpCombination($uid, $ip);
+	}
+
+	/**
+	 * @param string $token
+	 * @param string $ip
+	 * @return void
+	 */
+	public function addFailedLinkAccess($token, $ip) {
```

Comment:
I'm not sure if this is the right place for this method... I mean, anyone having access to the Throttle class could add a failed link access, which doesn't seem to be the responsibility of the class.

Reply:
the method removed from class as suggested.